src/library.js

```diff
@@ -3225,11 +3223,169 @@ mergeInto(LibraryManager.library, {
 #if MINIMAL_RUNTIME
     var f = dynCalls[sig];
 #else
+    #if MAIN_MODULE == 1
+    if (!('dynCall_' + sig in Module)) {
+      Module['dynCall_' + sig] = createDyncallWrapper(sig);
+    }
+    #endif
     var f = Module['dynCall_' + sig];
 #endif
     return args && args.length ? f.apply(null, [ptr].concat(args)) : f.call(null, ptr);
   },
   $dynCall__deps: ['$dynCallLegacy', '$getWasmTableEntry'],
+#if MAIN_MODULE == 1
+  $dynCallLegacy__deps: ['$createDyncallWrapper'],
+  $createDyncallWrapper__deps: ['$generateFuncType', '$uleb128Encode'],
+  $createDyncallWrapper: function(sig) {
+    var sections = [];
```

Comment:
This is really huge and unpleasant function.

I get that we might need it under certain circumstances but maybe we can take some steps to limits its impact.  How about moving it to its own file?  Maybe: `src/library_makeDynCall.js`
Is there some way we can have folks opt into it?  I'm loath to add a new setting.   Perhaps we could ask folks to do `-lmakeDynCall` or `--js-library=makeDynCall`?  (Let me see if either of those will work).